### ISSUE TITLE
Syslog-ng 1.6.12 consumes 70% cpu

### ISSUE THREAD

**Issue by karthik25101990** [balabit/syslog-ng#1290]
# syslog-ng
The syslog-ng-1.6.12 consumes 70% cpu when there are burst of incoming messages, ie when debug level messages are enabled. Due to this other processes dont get their share and the system reboots What are the default values  for log_fifo_size , log_fetch_limit etc ?? 

## Version of syslog-ng
1.6.12
output of syslog-ng --version
```

## Platform
Single core processor, 256 MB Ram
(Name and version of OS)
OpenWrt
# Issue
The syslog-ng-1.6.12 consumes 70% cpu when there are burst of incoming messages, ie when debug level messages are enabled. Due to this other processes dont get their share , What are the default values  for log_fifo_size , log_fetch_limit etc ??In syslog-ng.conf we have not set any of these things and
are using the 1.6 default conf file

## Failure

(Backtrace, error messages or detailed description of failure comes here.)
(To create debug bundles, use `syslog-debun`. Details of its usage can be found [here](https://www.balabit.com/sites/default/files/documents/syslog-ng-ose-latest-guides/en/syslog-ng-ose-guide-admin/html/syslog-debun.1.html))

```
$ gdb syslog-ng
 run
```

## Steps to reproduce

1. start syslog-ng

2. ...

## Configuration

```
$ cat /path/to/syslog-ng.conf

```

(or gist URL)

## Input and output logs (if possible)

**Comment by bazsi** [comment]
How many messages do you see per second? What kind of cpu is this? If its
openwrt then i guess its a MIPS cpu. Mhz? Is it running at 600mhz?

Do you have regexps in your config? That can decrease performance again.

I would say that 70% is realistic on a system like that with 1.6.12 with
1000-2000 messages per second.

On the other hand 1.6.12 is very old, is still the old non glib based
source that was rewritten 15 years ago.

So even if syslog-ng is the culprit on your system, we couldn't really help
but ask you to upgrade. Recent openwrt does have a newer version if I
remember correctly.

I'd recommend to upgrade the system, and troubleshoot the issue if still
present with the newer version.

HTH,
Bazsi


On Dec 23, 2016 06:52, "karthik25101990" <notifications@github.com> wrote:

syslog-ng

The syslog-ng-1.6.12 consumes 70% cpu when there are burst of incoming
messages, ie when debug level messages are enabled. Due to this other
processes dont get their share and the system reboots What are the default
values for log_fifo_size , log_fetch_limit etc ??
Version of syslog-ng

1.6.12
output of syslog-ng --version


## Platform
Single core processor, 256 MB Ram
(Name and version of OS)
OpenWrt
# Issue
The syslog-ng-1.6.12 consumes 70% cpu when there are burst of incoming
messages, ie when debug level messages are enabled. Due to this other
processes dont get their share , What are the default values  for
log_fifo_size , log_fetch_limit etc ??In syslog-ng.conf we have not
set any of these things and
are using the 1.6 default conf file

## Failure

(Backtrace, error messages or detailed description of failure comes here.)
(To create debug bundles, use `syslog-debun`. Details of its usage can
be found [here](https://www.balabit.com/sites/default/files/documents/syslog-ng-ose-latest-guides/en/syslog-ng-ose-guide-admin/html/syslog-debun.1.html))


$ gdb syslog-ng
run


## Steps to reproduce

1. start syslog-ng

2. ...

## Configuration


$ cat /path/to/syslog-ng.conf


(or gist URL)

## Input and output logs (if possible)


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/balabit/syslog-ng/issues/1290>, or mute the thread

**Comment by karthik25101990** [comment]
Yes it is based on openwrt. The only regexp used is the match filter

Is the options log_fetch_limit and iw_size and other performance tuning options 
available in v 1.6. ? I could see from the 1.6 reference manual that only below 
performance tuning options listed 
gc_idle_threshold()
gc_busy_threshold()
sync
log_fifo_size

There is no mention about the fetch_limit follow_freq parameters in 1.6 reference manual.
however these options are mentioned in the syslog-ng Admin guide (Edition 1.1.4 ) 

So are these options supported in v1.6 ? as adding these options to the cfg file gives parsing errors. And will fine tuning these parameters help improve the 
performance.

**Comment by Kokan** [comment]
I am closing this for two reason, because of inactivity and because of a really old syslog-ng version is not really supported.
